### ISSUE TITLE
Code mappings path is wrong

### ISSUE THREAD

**Issue by lforst** [getsentry/sentry#71216]
_Originally posted by @alessandro-newzoo in https://github.com/getsentry/sentry-javascript/issues/11865#issuecomment-2115344655_

_TLDR: The problem seems to be gone on new issues concerning the same JS file. But, even though Sentry shows the correct line and content of the source file, the path it's showing is wrong and if I click on it, it gives me a 404._

Hi, I'm the person this issue was opened for 👋🏻

@stayallive you're right, I'm working on our website's theme, and the content of the folder `wp-content/themes/newzoo/`  is the only part tracked by git and uploaded to GitHub, but the path mapping should be correct as other files within the same folder were mapping correctly.

Before this issue was opened there's been a long discussion between me and Stella via tickets, so you're all probably lacking a lot of context here.

Basically I had this issue in Sentry where `wp-content/themes/newzoo/js/_dist/single--games.bundle` failed to be mapped to its source file `wp-content/themes/newzoo/js/_src/single--games.js`: https://newzoo-com.sentry.io/share/issue/f56fc05f6b964da38d075cc4b110158f/

Other issues concerning files in the same folder as the above were mapping correctly, eg: https://newzoo-com.sentry.io/share/issue/00f6add9b4064aa685640939fa69ae87/

Although I'm only noticing now something odd. The two issues above show different paths to the source:

The issue that doesn't map, shows
`wp-content/themes/newzoo/js/_dist/single--games.bundle`

The issue that maps correctly, shows
`wp-content/themes/newzoo/js/_dist/newzoo/js/_src/single--trendreports`

↑ this one, even though is the one that works, shows a wrong path to the source:

```
wp-content/themes/newzoo/js/_dist/newzoo/js/_src/single--trendreports
                            ^^^^^^^^^^^^^^^^
                         this shouldn't be here
```

To sum it up,
the source files are at 
`wp-content/themes/newzoo/js/_src` 
the dist files at 
`wp-content/themes/newzoo/js/_dist`
and Sentry thinks the source files are at
`wp-content/themes/newzoo/js/_dist/newzoo/js/_src/`

So ironically the one that did not work was the one with the correct mapping.
It's somehow appending part of the src path to the full dist path.

I suspect that's why that was happening, though I have no idea why since both files were generated by the same webpack config. Is this something on`wp-sentry`'s end? Sentry's? Or something I need to fix?

Like I said above, I can see the correct content in the stack trace, but the path is not correct for some reason, so if I click on it I get a 404:

![CleanShotX - Arc  16 05 2024 at 15-31-11 @2x](https://github.com/getsentry/sentry-javascript/assets/47320294/74195893-5789-4ad8-a680-a67c7538a532)
![CleanShotX - Arc  16 05 2024 at 15-42-39](https://github.com/getsentry/sentry-javascript/assets/47320294/dbbab105-ea72-4649-b824-5e1d01717b0f)

I went ahead and deleted all Code Mappings, recreated it by clicking on `Set up Code Mapping` on the issue screen, which then created a new wrong mapping again (which works at showing the src content, but 404s on click):

https://github.com/getsentry/sentry-javascript/assets/47320294/39ddb4ca-1bfc-44b5-b26c-22e79cd63298

**Comment by jangjodi** [comment]
This is very odd behavior, @armenzg do you have any idea what might be happening here with the automatic code mappings?

**Comment by armenzg** [comment]
Hello @alessandro-newzoo for reporting.
The hyperlink for the JS file is unrelated to the code mappings feature (the GitHub icon on the right side) [1].

If you load the JSON event, you will see `abs_path` as the link to the JS file:
<img width="710" alt="image" src="https://github.com/getsentry/sentry/assets/44410/097e0cd0-556b-4558-9502-9c9f7ab8f042">

Please click the GitHub icon and let me know if that URL works as expected.

[1] Hyperlink for JS files and GitHub links
![image](https://github.com/getsentry/sentry/assets/44410/9fed7a02-cb23-4dbe-8126-0b62e6b57dc4)

**Comment by lforst** [comment]
@alessandro-newzoo can you share the content of the `sources` field in your source map? I think that is generally where that path comes from.

**Comment by lforst** [comment]
@armenzg did you try out using `raw_stacktrace.frames[].abs_path`?

**Comment by armenzg** [comment]
@alessandro-newzoo we are wondering if this link causes more confusion than it helps:
<img width="136" alt="image" src="https://github.com/getsentry/sentry/assets/44410/044c2030-6e9d-4a2c-84eb-bfaaaaa6f5b6">
since you have access to the source file via the GitHub icon:
<img width="31" alt="image" src="https://github.com/getsentry/sentry/assets/44410/31e598da-22db-4b71-b86e-dd5a3cf26280">

The first hyperlink is mostly useful if there's no GitHub link.

**Comment by alessandro-newzoo** [comment]
Hi,

personally, I think they serve two different purposes and complement each other:
- it's nice to double check the file exists/is in the right path on the live server;
- the GitHub icon takes you to the source file, whereas the link takes you to the compiled file;
- the repo doesn't contain the dist files as they're compiled on the CI, so this is the most practical way to see their content if needed.

None of these points are crucial, but they're nice to have.

Thank you

**Comment by loewenheim** [comment]
Hi, is this issue still relevant? I haven't had any luck reconstructing the problem; the link at [https://newzoo-com.sentry.io/share/issue/f56fc05f6b964da38d075cc4b110158f/](https://newzoo-com.sentry.io/share/issue/f56fc05f6b964da38d075cc4b110158f/) works but the event doesn't fit with the description, and the link at [https://newzoo-com.sentry.io/issues/5141223488](https://newzoo-com.sentry.io/issues/5141223488) doesn't work (expired, I assume).

**Comment by loewenheim** [comment]
I'm closing this as inactive. Please feel free to reopen if necessary.
